src/lm/arpa-file-parser.h

```diff
@@ -86,9 +91,35 @@ class ArpaFileParser {
   void Read(std::istream &is, bool binary);
 
   /// Parser options.
-  const ArpaParseOptions& Options() const { return options_; }
+  const ArpaParseOptions &Options() const { return options_; }
 
  protected:
+  /// A status value returned from ConsumeNGram() overload.
+  class Status {
+   public:
+    /// Return the warninig status to print a warninig and continue.
+    static Status Warning(const std::string& message) {
+      return Status(kWarninig, message); }
```

Comment:
Or perhaps there's some other reason why you're doing this, like giving the warning more context because ProcessNGram (or whatever) doesn't know the line number.  I don't think a huge amount of context (like line numbers) is not really necessary because it's not hard to work out from the ARPA file, and these errors are rare occurrences.  I'd be happier you just using KALDI_ERR and KALDI_WARN (guarded with a counter), called directly where it's needed, rather than this HandleError thing.  That makes the code much easier to debug as the error happens where it originated and all the variables are on the stack; and people familiar with the Kaldi coding style will understand what's going on.